src/rgb/contract/assignments.rs

```diff
@@ -1039,83 +1038,167 @@ mod test {
             }
+            let input_sum: u64 = input_amounts.iter().sum();
 
             // Create an output amount vector such that
             // input.sum() = output.sum(), but
             // input.count() != output.count()
-            // We do this by generating an empty output vector
-            // of random size. Then randomly add 1 to random positions
-            // until the sum of the vector matches with the input sum
-            // This also creates 0 value allocations probabilistically
 
-            let input_sum: u64 = amounts.iter().sum();
             let mut output_amounts = vec![0u64; rng.gen_range(1, 20)];
             let output_length = output_amounts.len();
-            for _ in 0..input_sum {
-                output_amounts[rng.gen_range(0, output_length)] += 1u64;
+
+            // Add random values to output amounts until the last element
+            for index in 0..output_length - 1 {
+                output_amounts[index] = rng.gen_range::<u64>(100_000, 100_000_000_000);
+            }
+            let output_sum: u64 = output_amounts.iter().sum();
+
+            // Balance input and output amount vector based on their sums
+            if output_sum > input_sum {
+                input_amounts[input_length - 1] += output_sum - input_sum;
+            } else {
+                output_amounts[output_length - 1] += input_sum - output_sum;
             }
 
             // Check if test passes
             assert!(compute_zero_balance(
-                &amounts[..],
+                &input_amounts[..],
                 &output_amounts[..],
                 rng.gen_range(0, output_length)
             ));
+
+            //Check non-equivalent amounts do not verify
+            assert!(!compute_zero_balance(
+                &input_amounts[..(input_length - 1)],
+                &output_amounts[..(output_length - 1)],
+                rng.gen_range(0, output_length)
+            ));
         }
 
-        // Test when input.sum() != output.sum()
-        // When they only differ by 1
-        // When they differ by u64::MAX
-        assert!(!compute_zero_balance(&amounts[1], &amounts[2], 2));
-        assert!(!compute_zero_balance(&[1, 2, 3, 4], &[1, 2, 3, 5], 2));
-        assert!(!compute_zero_balance(
-            &[1, 2, 3, 0],
-            &[1, 2, 3, u64::MAX],
-            2
+        // Test Overflow conditions
+        assert!(compute_zero_balance(&[u64::MAX + 1], &[0], 1));
+        assert!(compute_zero_balance(
+            &[u64::MAX + u64::MAX],
+            &[u64::MAX - 1],
+            1
+        ));
+        assert!(compute_zero_balance(
+            &[(u32::MAX * u32::MAX - 1) as u64],
+            &[0],
+            1
+        ));
+        assert!(compute_zero_balance(
+            &[(u32::MAX * u32::MAX) as u64],
+            &[1],
+            1
+        ));
+        assert!(compute_zero_balance(
+            &[(u32::MAX * u32::MAX + 1) as u64],
+            &[2],
```

Comment:
```suggestion
            &[u64::MAX * u64::MAX + 1],
            &[0],
```

Reply:
I am trying to update this. For some reason, the test went really slow and taking over 5 mins. Also not sure why Travis is failing. Will look into these issues tomorrow.